mm.lic

```diff
@@ -0,0 +1,94 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#mm
+=end
+custom_require.call(%w[common-arcana events])
+
+class MoonMagery
+  include DRCA
+  def initialize
+    args = get_args
+    @settings = get_settings
+
+    if args.locate
+      locate?(args.person)
+    elsif args.rift
+      rift(args.person)
+    end
+    Flags.delete('locate-fail')
+  end
+
+  def get_args
+    arg_defs = [
+      [
+        { name: 'locate', regex: /locate/i },
+        { name: 'person', regex: /\w+/, description: 'Name of person to locate' }
+      ],
+      [
+        { name: 'rift', regex: /rift/i },
+        { name: 'person', regex: /\w+/, description: 'Name of person to rift' }
+      ]
+    ]
+
+    parse_args(arg_defs)
+  end
+
+  def rift(person)
+    person = person.capitalize
+    check_moonwatch
+    unless UserVars.moons.any? { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 0 }
+      echo 'No moons available to cast Rift'
+      return
+    end
+    data = if @settings.waggle_sets['rs']
+             @settings.waggle_sets['rs'].values.first
+           else
+             echo "Using discern values for RS. Make a waggle_set called 'rs' to avoid this in the future"
+             check_discern({ 'abbrev' => 'RS' }, @settings)
+           end
+    unless locate?(person)
+      echo "Unable to rift #{person} due to failed locate"
+      return
+    end
+    data['cast'] = "cast #{person}"
+    cast_spell(data, @settings)
+  end
+
+  def locate?(person)
+    person = person.capitalize
+    Flags.add('locate-fail', 'You see a very vague vision of', "The details of #{person}'s location are unclear", 'Your vision is not clear enough for your spell to succeed.', 'There doesn\'t seem to be anyone by that name under the skies')
+    data = if @settings.waggle_sets['locate']
+             @settings.waggle_sets['locate'].values.first
+           else
+             echo "Using discern values for locate. Make a waggle_set called 'locate' to avoid this in the future"
+             check_discern({ 'abbrev' => 'locate' }, @settings)
+           end
+    data['cast'] = "cast #{person}"
+    cast_spell(data, @settings)
+    pause 0.25
+    if Flags['locate-fail']
+      case Flags['locate-fail'].first
+      when /Your vision is not clear/
+        return locate?(person)
```

Comment:
I'm not too familiar with MMs. Is it possible for this to loop forever, or is it very likely the locate will work on the next try?

Reply:
It will loop forever if they fail to cast clear vision in 5 seconds (as `cast_spell` does)

Reply:
Ill have the script exit rather than retry.